db/migrate/20221130150352_create_project_journals.rb

```diff
@@ -0,0 +1,15 @@
+class CreateProjectJournals < ActiveRecord::Migration[7.0]
+  # rubocop:disable Rails/CreateTableWithTimestamps(RuboCop)
+  def change
+    create_table :project_journals do |t|
+      t.string :name, default: '', null: false
+      t.text :description
+      t.boolean :public, default: true, null: false
+      t.bigint :parent_id
+      t.string :identifier
+      t.boolean :active, default: true, null: false
+      t.boolean :templated, default: false, null: false
+    end
+  end
+  # rubocop:enable Rails/CreateTableWithTimestamps(RuboCop)
```

Comment:
We'd need a data migration as well to fill up the table initially for those projects that already exist. More or less a 
```
INSERT INTO project_journals ... SELECT ... FROM projects
```
(and of course the `journals` entries.)

As for the user associated to that, we could use `User.system` for it since there isn't any better value, I think.

Reply:
Done in 373739e

It was not particularly easy, as the id of the `project_journals` has to be known to set the `journals.data_id` value, but there is no `project_id` in `project_journals` to be able to make a join between `journals`, `projects`, and the `project_journals` data. I used `identidfier` as it has a uniq contraint on it.

Please for review pay attention to naming: some things are prefixed with "project_attributes", some other are prefixed "projects". I have the feeling that everything should be named "project_attributes" or "projects". Please tell me what you think about it.